File: ios/MendixSplashScreen/MendixSplashScreen.m
//
//  Copyright (c) Mendix, Inc. All rights reserved.
//
#import "ReactNative.h"
#import "react-native-splash-screen/RNSplashScreen.h"

@interface MendixSplashScreen : NSObject<RCTBridgeModule>
@end

@implementation MendixSplashScreen

RCT_EXPORT_MODULE()

RCT_EXPORT_METHOD(show) {
  [ReactNative.instance showSplashScreen];
}

RCT_EXPORT_METHOD(hide) {
  [ReactNative.instance hideSplashScreen];
}

@end

File: ios/Modules/MendixSplashScreen/MendixSplashScreen.swift
import Foundation

@objc public protocol SplashScreenPresenterProtocol: AnyObject {
    @objc func show(_ rootView: UIView?)
    @objc func hide()
}

@objcMembers public class MendixSplashScreen: NSObject {
    
    public func show() {
        ReactNative.shared.showSplashScreen()
    }
    
    public func hide() {
        ReactNative.shared.hideSplashScreen()
    }
}

//Checked
